src/AWS.Deploy.Orchestration/CustomRecipeLocator.cs

```diff
@@ -0,0 +1,176 @@
+        private readonly IOrchestratorInteractiveService _orchestratorInteractiveService;
+        private readonly ICommandLineWrapper _commandLineWrapper;
+        private readonly IDeploymentManifestEngine _deploymentManifestEngine;
+        private readonly IDirectoryManager _directoryManager;
+        
+        public CustomRecipeLocator(IDeploymentManifestEngine deploymentManifestEngine, IOrchestratorInteractiveService orchestratorInteractiveService,
+            ICommandLineWrapper commandLineWrapper, IDirectoryManager directoryManager)
+        {
+            _orchestratorInteractiveService = orchestratorInteractiveService;
+            _commandLineWrapper = commandLineWrapper;
+            _deploymentManifestEngine = deploymentManifestEngine;
+            _directoryManager = directoryManager;
+        }
+
+        /// <summary>
+        /// Wrapper method to fetch fetch custom recipe paths from a deployment-manifest file as well as
+        /// other locations that are monitored by the same source control root as the target application that needs to be deployed.
+        /// </summary>
+        /// <param name="targetApplicationFullPath">The absolute path to the csproj or fsproj file of the target application</param>
+        /// <returns>A <see cref="HashSet{String}"/> containing absolute paths of directories inside which the custom recipe snapshot is stored</returns>
+        public async Task<HashSet<string>> LocateCustomRecipePaths(string targetApplicationFullPath)
+        {
+            await LocateRecipePathsFromManifestFile(targetApplicationFullPath);
+
+            await LocateAlternateRecipePaths(targetApplicationFullPath);
+            
+            return new HashSet<string>(_customRecipePaths);
+        }
+
+        /// <summary>
+        /// Fetches custom recipe paths by parsing the deployment-manifest file that is associated with the target application.
+        /// </summary>
+        /// <param name="targetApplicationFullPath">The absolute path to the target application csproj or fsproj file</param>
+        private async Task LocateRecipePathsFromManifestFile(string targetApplicationFullPath)
+        {
+            List<string>? manifestFileRecipePaths;
+            try
+            {
+                manifestFileRecipePaths = await _deploymentManifestEngine.GetCustomRecipePaths(targetApplicationFullPath);
+            }
+            catch
+            {
+                _orchestratorInteractiveService.LogMessageLine(Environment.NewLine);
+                _orchestratorInteractiveService.LogErrorMessageLine("Failed to load custom deployment recommendations " +
+                   "from the deployment-manifest file due to an error while trying to deserialze the file.");
+                return;
+            }
+
+            foreach (var recipePath in manifestFileRecipePaths)
+            {
+                _customRecipePaths.Add(recipePath);
```

Comment:
Validate the `recipePath` still points to valid recipe.

Reply:
Fixed